protocols/relay/src/priv_client.rs

```diff
@@ -79,6 +80,10 @@ pub struct Behaviour {
     /// connection.
     directly_connected_peers: HashMap<PeerId, Vec<ConnectionId>>,
 
+    relay_connection_addr: HashMap<ConnectionId, Multiaddr>,
+
+    reservation: HashSet<Multiaddr>,
```

Comment:
I am not quite happy yet with these data structures.

For one, `relay_connection_addr` actually stores "pending" reserverations, right?
Two, `reservation` stores "confirmed" reservations.

We could combine them into one data structure and use an enum to say whether it is pending or confirmed.

Reply:
Something like:

```rust
/// Stores the address of a pending or confirmed reservation.
///
/// This is indexed by the [`ConnectionId`] to a relay server and the address is the `/p2p-circuit` address we reserved on it.
reservation_addresses: HashMap<ConnectionId, (Multiaddr, ReservationStatus)>

// ---

enum ReservationStatus {
	Pending,
	Confirmed
}
```

Reply:
We could also create a wrapper around this `HashMap` to make usage a bit more concise.

Reply:
Thats fine with me. I was considering on using the current hash map, but wasnt sure if that would be desired, but it is a simple change to do :)